WordPress/gutenberg
# Docgen: split `getIntermediateRepresentation` unit tests in smaller tests

## Description

The [`getIntermediateRepresentation` unit tests](https://github.com/WordPress/gutenberg/blob/trunk/packages/docgen/test/get-intermediate-representation.js) currently have a lot of assertions per test, making them hard to read and debug in case of failure.
This PR aims at splitting the assertion into smaller tests.

The actual code for the tests hasn't been updated. The main changes are:
- Add one intermediate level of `describe` around most tests
- Split long tests into smaller ones

Most of the diff changes are caused by the change in indentation.

## How has this been tested?

- Run `npm run test-unit packages/docgen`
- Make sure that all tests are passing
- Check how the list of tests for `getIntermediateRepresentation` is more granular

## Screenshots 

N/A

## Types of changes





Refactor

## Checklist:
- [ ] (N/A) I've tested my changes with keyboard and screen readers.
